nrf: clock reimplementation breaks test

`tests/arch/arm/arm_irq_vector_table` fails to build on nrf52_pca20020, nrf52_pca10040, and nrf52840_pca10056 starting at 6700f2f1940c645c5cd1eb23baad8bd0c8041b81 with this error:
```
make[3]: Leaving directory '/scratch/zephyr/zp1/zephyr/sanity-out/nrf52_pca20020/tests/arch/arm/arm_irq_vector_table/arch.interrupt'
make[3]: Entering directory '/scratch/zephyr/zp1/zephyr/sanity-out/nrf52_pca20020/tests/arch/arm/arm_irq_vector_table/arch.interrupt'
[ 98%] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj
[100%] Linking C executable zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       25964 B       512 KB      4.95%
            SRAM:        5164 B        64 KB      7.88%
        IDT_LIST:          0 GB         2 KB      0.00/usr/local/zephyr-sdk-0.10.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: ../app/libapp.a(arm_irq_vector_table.c.obj):/scratch/zephyr/zp1/zephyr/tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:167: undefined reference to `nrf_power_clock_isr'
collect2: error: ld returned 1 exit status
%
make[3]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:101: zephyr/zephyr.elf] Error 1
```

This [breaks CI checking](https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/52738/16/console) of #19650.

--- a/drivers/clock_control/nrf_power_clock.c
+++ b/drivers/clock_control/nrf_power_clock.c
@@ -187,7 +187,15 @@ static int clock_start(struct device *dev, clock_control_subsys_t sub_system)
 	return clock_async_start(dev, sub_system, NULL);
 }
 
-static void nrf_power_clock_isr(void *arg);
+/* Note: this function has public linkage, and MUST have this
+ * particular name.  The platform architecture itself doesn't care,
+ * but there is a test (tests/kernel/arm_irq_vector_table) that needs
+ * to find it to it can set it in a custom vector table.  Should
+ * probably better abstract that at some point (e.g. query and reset
+ * it by pointer at runtime, maybe?) so we don't have this leaky
+ * symbol.
+ */
+void nrf_power_clock_isr(void *arg);
 
 static int hfclk_init(struct device *dev)
 {
@@ -316,15 +324,6 @@ static void usb_power_isr(void)
 #endif
 }
 
-
-/* Note: this function has public linkage, and MUST have this
- * particular name.  The platform architecture itself doesn't care,
- * but there is a test (tests/kernel/arm_irq_vector_table) that needs
- * to find it to it can set it in a custom vector table.  Should
- * probably better abstract that at some point (e.g. query and reset
- * it by pointer at runtime, maybe?) so we don't have this leaky
- * symbol.
- */
 void nrf_power_clock_isr(void *arg)
 {
 	ARG_UNUSED(arg);

